src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java

```diff
@@ -182,9 +184,18 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         .orElseGet(() -> new Module[0]);
     ModuleSelectionViewModel moduleSelectionViewModel
         = new ModuleSelectionViewModel(modules, project);
-    if (!moduleDialogFactory.createDialog(moduleSelectionViewModel).showAndGet()
-        || moduleSelectionViewModel.getSelectedModule() == null) {
-      return;
+    String moduleName = tryGetExerciseModuleName(course, exercise);
+    if (moduleName == null) {
+      if (!moduleDialogFactory.createDialog(moduleSelectionViewModel).showAndGet()
+          || moduleSelectionViewModel.getSelectedModule() == null) {
+        return;
+      }
+    } else {
+      moduleSelectionViewModel.setSelectedModule(moduleName);
```

Comment:
Nit: do we need to use a view model here if we don't have any view?

Reply:
Also: pretty complex logic here... perhaps `Menu Key` + `R`? ;)

Reply:
Hmm we don't, good point